Release notes for Docker Desktop 4.20 (#17396)
docker/docker.github.io: desktop/release-notes.md

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.20.0

{% include release-date.html date="2023-05-25" %}

{% include desktop-install.md all=true version="4.20.0" build_path="/" %}

### New

- [Docker Scout CLI](https://docs.docker.com/scout/#docker-scout-cli) now finds the most recently built image if it is not provided as an argument.
- Improved the [Docker Scout CLI](https://docs.docker.com/scout/#docker-scout-cli) `compare` command.

### Upgrades

- [Buildx v0.10.5](https://github.com/docker/buildx/releases/tag/v0.10.5)
- [Compose v2.18.1](https://github.com/docker/compose/releases/tag/v2.18.1)

> - [Docker Engine v23.0.5](https://docs.docker.com/engine/release-notes/23.0/#2305)

Reviewer: This should be `24.0.1`

- [Containerd v1.6.21](https://github.com/containerd/containerd/releases/tag/v1.6.21)
- [runc v1.1.7](https://github.com/opencontainers/runc/releases/tag/v1.1.5)

> - [runc v1.1.6](https://github.com/opencontainers/runc/releases/tag/v1.1.6)

Reviewer 2: Duplicate

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.20.0

{% include release-date.html date="2023-05-25" %}

> {% include desktop-install.md all=true version="4.20.0" build_path="/" %}

Reviewer 3: To-do: Update the checksums 
(Although I know this is already on your radar as it can only be done last minute)

Reviewer 3: @lorenrh are these ready to be added yet?

### New

> - [Docker Scout CLI](https://docs.docker.com/scout/#docker-scout-cli) find the most recently built image if none has been provided as an argument.

Reviewer 3: ```suggestion
- [Docker Scout CLI](https://docs.docker.com/scout/#docker-scout-cli) now finds the most recently built image if it is not provided as an argument.
```

### Upgrades

- [Buildx v0.10.5](https://github.com/docker/buildx/releases/tag/v0.10.5)
- [Compose v2.18.1](https://github.com/docker/compose/releases/tag/v2.18.1)
- [Docker Engine v23.0.5](https://docs.docker.com/engine/release-notes/23.0/#2305)
- [Containerd v1.6.21](https://github.com/containerd/containerd/releases/tag/v1.6.21)
- [runc v1.1.7](https://github.com/opencontainers/runc/releases/tag/v1.1.5)
- [runc v1.1.6](https://github.com/opencontainers/runc/releases/tag/v1.1.6)

### Bug fixes and enhancements

#### For all platforms

- Added a warning about the [retirement of Docker Compose ECS/ACS integrations in November 2023](https://docs.docker.com/go/compose-ecs-eol/). Can be suppressed with `COMPOSE_CLOUD_EOL_SILENT=1`.

> - [Docker Scout CLI](https://docs.docker.com/scout/#docker-scout-cli) find the most recently built image if none has been provided as an argument.

Reviewer 3: This can be removed as it's repeated above

> - Improved [Docker Scout CLI](https://docs.docker.com/scout/#docker-scout-cli) `compare` command.

Reviewer 3: ```suggestion
- Improved the [Docker Scout CLI](https://docs.docker.com/scout/#docker-scout-cli) `compare` command.
```

Author: this line I've also moved it to the new section. It's more of a personal opinion here to put Scout CLI in the _new_ section! So let me know if I should move it to the _bug fixes and enhancements_ section!

Reviewer 3: I think we should keep it in Bug fixes and enhancements. Looking at previous releases, we've generally kept all the 'improved..' notes, in this section and it would be good to keep this consistent

> - Fix HTTP proxy bug where an HTTP 1.0 client could receive an HTTP 1.1 response.

Reviewer 3: ```suggestion
- Fixed an HTTP proxy bug where an HTTP 1.0 client could receive an HTTP 1.1 response.
```

> - Enabled Docker Desktop's Enhanced Container Isolation (ECI) feature on WSL-2 (Docker Business only).

Reviewer 3: ```suggestion
- Enabled Docker Desktop's Enhanced Container Isolation (ECI) feature on WSL-2. This is available with a Docker Business subscription.
```

> - Fixed a bug on the containers list table where hiding columns would not be persisted for a fresh installation of Docker Desktop

Reviewer 3: ```suggestion
- Fixed a bug on the **Containers** table where previously hidden columns were displayed again after a fresh installation of Docker Desktop.
```

#### For Mac

> - Reclaim disk space more quickly when files are deleted in containers. Related to [docker/for-mac#371](https://github.com/docker/for-mac/issues/371).

Reviewer 3: ```suggestion
- You can now reclaim disk space more quickly when files are deleted in containers. Related to [docker/for-mac#371](https://github.com/docker/for-mac/issues/371).
```

For frequently asked questions about Docker Desktop releases, see [FAQs](faqs/general.md/#releases)

## 4.20.0

{% include release-date.html date="2023-05-25" %}

> {% include desktop-install.md all=true version="4.20.0" build_path="/" %}

Author: 😮‍💨 😶‍🌫️ Thanks!
